Better handling of processing 3d tiles

lilleyse opened the pull request:
While running a Draco compressed tileset we noticed that tiles seems to be streaming in reverse order. The problem was we were iterating over the processing queue in reverse order, mainly as a convenience since elements could be removed during iteration. In this tileset downloading tiles is much quicker than processing so the processing queue was stuffed with tiles and kept on processing the low priority tiles first.

The fix was to iterate in forward order and afterwards filter out tiles that are done processing.

Before/after:
![nyc](https://user-images.githubusercontent.com/915398/37837836-b7e84a66-2e8c-11e8-9961-cecfb4e1d7d7.gif)
![nyc-better](https://user-images.githubusercontent.com/915398/37838028-3d1a8a3c-2e8d-11e8-8098-0a058c6fe4b2.gif)

cesium-concierge commented:
![Signed CLA is on file.](https://img.shields.io/badge/CLA-signed-brightgreen.svg)

@lilleyse, thanks for the pull request! Maintainers, we have a signed CLA from @lilleyse, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

lilleyse commented:
I'm not sure why 3 of the tests fail in travis but not locally... I'll keep looking.

pjcozzi commented:
Update CHANGES.md and add a unit test if possible.  Are you also porting this to your refactor?

lilleyse commented:
Fixed the tests. One of the files in the i3dm specs data folder was called `Box.glb` instead of `box.glb`. The output from 3d-tiles-tools/sampler-generator looks fine so I'm not sure how that happened. The code here must have exposed that problem in some way.

I also updated CHANGES.md but didn't write a new test because it was too dependent on load order and the test started to get complicated.

> Are you also porting this to your refactor

Yeah these changes will fit in fine there.

lilleyse commented:
@likangning93 given that you are slated to review some of the other 3D Tiles PRs maybe it makes sense to review this too?

lilleyse commented on Source/Scene/Cesium3DTileset.js:
```diff
@@ -1510,13 +1509,31 @@ define([
+                tiles[i - removeCount] = tile;
+            }
+        }
+        tiles.length -= removeCount;
```
For whoever is reviewing, this code is based on a similar filter function in [RequestScheduler](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/RequestScheduler.js#L220-L243)

lilleyse commented:
Ah I forgot @likangning93 is out today - @bagnell could you review today?
